internal/status/gateway.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	api_equality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gateway_v1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+)
+
+// SyncGateway computes the current status of gw and updates status based on
+// any changes since last sync.
+func SyncGateway(ctx context.Context, cli client.Client, gw *gateway_v1alpha1.Gateway, errs field.ErrorList) error {
+	updated := gw.DeepCopy()
+
+	// Gateway's contain a default status condition that must be removed when reconciled by a controller.
```

Comment:
I'm pretty sure this is incorrect, it should be set to `status: true` when reconciled by a controller.

Reply:
Ack